frontend/src/metabase-lib/parameters/utils/targets.ts

```diff
@@ -37,20 +43,30 @@ export function getParameterTargetField(
   return null;
 }
 
-export function buildDimensionTarget(dimension: Dimension) {
-  return ["dimension", dimension.mbql()];
+export function buildDimensionTarget(
+  dimension: Dimension,
+): NativeParameterDimensionTarget {
+  return [
+    "dimension",
+    dimension.mbql() as DimensionReference,
+  ] as NativeParameterDimensionTarget;
 }
 
 export function buildColumnTarget(
   query: Lib.Query,
   stageIndex: number,
   column: Lib.ColumnMetadata,
-) {
-  return ["dimension", Lib.legacyRef(query, stageIndex, column)];
+): StructuredParameterDimensionTarget {
+  return [
+    "dimension",
+    Lib.legacyRef(query, stageIndex, column),
+  ] as StructuredParameterDimensionTarget;
```

Comment:
We can drop the cast if we introduce a type guard `const isLocalFieldReference = (value: FieldReference): value is LocalFieldReference`:


```suggestion
  const fieldReference = Lib.legacyRef(query, stageIndex, column);

  if (!isLocalFieldReference(fieldReference)) {
    throw new Error("...");
  }

  return ["dimension", fieldReference];
```

There might be a better way, but I could not find it.

Reply:
[a2d99a1](https://github.com/metabase/metabase/pull/38596/commits/a2d99a16c11199027587bd8f7f1083f0a0c8e17e)